doc/io_streams.rdoc

```diff
@@ -326,23 +396,112 @@ that determine how lines in a stream are to be treated:
 
-=== Open and Closed \IO Streams
+=== Character \IO
 
-A new \IO stream may be open for reading, open for writing, or both.
+You can process an \IO stream character-by-character using these methods:
 
-You can close a stream using these methods:
+- IO#getc: Reads and returns the next character from the stream:
 
-- IO#close: Closes the stream for both reading and writing.
-- IO#close_read (not available in \ARGF): Closes the stream for reading.
-- IO#close_write (not available in \ARGF): Closes the stream for writing.
+    f = File.new('t.rus')
+    f.getc # => "т"
+    f.getc # => "е"
+    f.getc # => "с"
+    f.getc # => "т"
+    f.getc # => nil
 
-You can query whether a stream is closed using these methods:
+- IO#readchar (not in \StringIO):
+  Like #getc, but raises an exception at end-of-stream:
 
-- IO#closed?: Returns whether the stream is closed.
+    f.readchar # Raises EOFError.
+
+- IO#ungetc (not in \ARGF):
+  Pushes back ("unshifts") a character or integer onto the stream:
+
+    f = File.new('t.rus')
+    c = f.getc         # => "т"
+    f.ungetc(c)
+    f.ungetc(c)
+    f.read             # => "ттест"
+    f.close
+    File.read('t.rus') # => "тест"
+
+- IO#putc (also in Kernel): Writes a character to the stream:
+
+    c = File.new('t.rus').getc # => "т"
+    f = File.new('t.tmp', 'w')
+    f.putc(c)
+    f.putc(c)
+    f.close
+    File.read('t.tmp')         # => "тт"
+
+- IO#each_char: Reads each remaining character in the stream,
+  passing the character to the given block:
+
+    f = File.new('t.rus')
+    f.pos = 4
+    f.each_char {|c| p c }
+
+  Output:
 
-=== Stream End-of-File
+    "с"
+    "т"
 
-You can query whether a stream is at end-of-file using this method:
+=== Byte \IO
+
+You can process an \IO stream byte-by-byte using these methods:
+
+- IO#getbyte: Returns the next 8-bit byte as an integer in range 0..255:
+
+    File.read('t.dat')
+    # => "\xFE\xFF\x99\x90\x99\x91\x99\x92\x99\x93\x99\x94"
+    File.read('t.dat')
+    # => "\xFE\xFF\x99\x90\x99\x91\x99\x92\x99\x93\x99\x94"
+    f = File.new('t.dat')
+    f.getbyte # => 254
+    f.getbyte # => 255
+    f.seek(-2, :END)
+    f.getbyte # => 153
+    f.getbyte # => 148
+    f.getbyte # => nil
+
+- IO#readbyte (not in \StringIO):
+  Like #getbyte, but raises an exception if at end-of-stream:
+
+    f.readbyte # Raises EOFError.
+
+- IO#ungetbyte (not in \ARGF):
+  Pushes back ("unshifts") a byte back onto the stream:
+
+    f.ungetbyte(0)
+    f.ungetbyte(01)
+    f.read # => "\u0001\u0000"
+
+- IO#each_byte: Reads each remaining byte in the stream,
+  passing the byte to the given block:
+
+    f.seek(-4, :END)
+    f.each_byte {|b| p b }
+
+  Output:
+
+    153
+    147
+    153
+    148
+
+=== Codepoint \IO
+
+You can process an \IO stream codepoint-by-codepoint using method
++#each_codepoint+:
+
+  f = File.new('t.rus')
+  a = []
+  f.each_codepoint {|c| a << c }
+  a # => [1090, 1077, 1089, 1090]
+  f.close
 
+<<<<<<< HEAD
```

Comment:
You have an unresolved git conflict here. I think you're missing part of the `Open and Closed IO Streams` section.

Reply:
Removed.